product/cmd/grpc/client/main_test.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+
+	"product/internal/grpc/domain"
+	domainmock "product/internal/grpc/domain/mock_domain"
+)
+
+var product = &domain.Product{
+	Id:          1,
+	Name:        "Sherlock Holmes",
+	Description: "It is an investigative book",
+	Price:       123.56,
+	Slug:        "sherlock-holmes",
+	Quantity:    60,
+	CreatedAt:   &domain.Timestamp{Timestamp: ptypes.TimestampNow()},
+}
+
+var req = &domain.AddProductRequest{
+	Product: product,
+}
+
+var resp = &domain.AddProductResponse{
+	CreatedProduct: product,
+}
```

Comment:
Declare these inside the function in which they're used. Globals are bad, but globals in tests are even worse.

Reply:
Thank you Arran, I have implemented the feedback, I'll probably give `Bazel` a study later